broker/src/main/java/org/apache/rocketmq/broker/transaction/queue/TransactionalMessageServiceImpl.java

```diff
@@ -187,6 +190,39 @@ public void check(long transactionTimeout, int transactionCheckMax,
                             }
                         }
 
+                        if (BrokerRole.SLAVE.equals(this.transactionalMessageBridge.getBrokerController().getMessageStoreConfig().getBrokerRole())
+                            && this.transactionalMessageBridge.getBrokerController().getMinBrokerIdInGroup()
+                            == this.transactionalMessageBridge.getBrokerController().getBrokerIdentity().getBrokerId()
+                            && this.transactionalMessageBridge.getBrokerController().getBrokerConfig().isEnableSlaveActingMaster()
```

Comment:
“EnableSlaveActingMaster” is default set to false. maybe it's better to let this condition checked at first.

Reply:
new order:

1. check EnableSlaveActingMaster first for default false;
2. check brokerId for only 1 broker meeting this condition in a cluster;
3. check broker role.